app/scripts/controllers/replicaSet.js

```diff
@@ -445,7 +446,9 @@ angular.module('openshiftConsole')
             $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][rsName] = replicaSet;
           } else if (action === 'MODIFIED') {
             // After the deployment ends remove him from the deploymentConfigDeploymentsInProgress structure.
-            delete $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][rsName];
+            if($scope.deploymentConfigDeploymentsInProgress && $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName]) {
```

Comment:
pretty sure `$scope.deploymentConfigDeploymentsInProgress` will always be defined, its going to get defined during the initial data returned by the watch.  were you seeing a case where it wasn't? because that would break line 445 as well.

My guess is the problem is just `$scope.deploymentConfigDeploymentsInProgress[deploymentConfigName]` not being defined